spring-integration-redis/src/test/java/org/springframework/integration/redis/outbound/ReactiveRedisStreamMessageHandlerTests.java

```diff
@@ -0,0 +1,223 @@
+		assertThat(record.getStream()).isEqualTo(streamKey);
+		assertThat(record.getValue()).isEqualTo(messagePayload);
+		template.delete(streamKey).block();
+	}
+
+	//TODO Find why the deserialization fail does not work
+	/*@Test
+	@RedisAvailable*/
+	public void explicitJsonSerializationContextTest() {
+		String streamKey = "myStream";
+		List<String> messagePayload = Arrays.asList("Bonjour", "à", "tous", "les", "confinés");
+
+		RedisSerializationContext<String, Object> jsonSerializationContext = redisStringOrJsonSerializationContext(false, List.class);
+
+		ReactiveRedisStreamMessageHandler streamMessageHandler = new ReactiveRedisStreamMessageHandler(streamKey, redisConnectionFactory);
+		streamMessageHandler.setSerializationContext(jsonSerializationContext);
+		//initializes reactiveRedisStreamOperations
+		invokeOnInitMethod(streamMessageHandler);
+
+		ReactiveMessageHandlerAdapter handlerAdapter = new ReactiveMessageHandlerAdapter(streamMessageHandler);
+		handlerAdapter.handleMessage(new GenericMessage<>(messagePayload));
+
+		ReactiveRedisTemplate<String, String> template = new ReactiveRedisTemplate(redisConnectionFactory,
+				jsonSerializationContext);
+		ObjectRecord<String, List> record = template.opsForStream().read(List.class, StreamOffset.fromStart(streamKey))
+				.blockFirst();
+
+		assertThat(record.getStream()).isEqualTo(streamKey);
+		assertThat(record.getValue()).isEqualTo("[\"Bonjour\", \"à\", \"tous\", \"les\", \"confinés\"]");
+		template.delete(streamKey).block();
+	}
+
+	//TODO Find why the deserialization does not work
+	/*@Test
+	@RedisAvailable*/
+	public void explicitJsonSerializationContextWithModelTest() {
+		String streamKey = "myStream";
+
+		RedisMessageStoreTests.Address address = new RedisMessageStoreTests.Address().withAddress("Rennes, France");
+		RedisMessageStoreTests.Person person = new RedisMessageStoreTests.Person(address, "Attoumane");
+
+		Message message = new GenericMessage(person);
+
+		RedisSerializationContext<String, Object> jsonSerializationContext = redisStringOrJsonSerializationContext(false, RedisMessageStoreTests.Person.class);
+
+		ReactiveRedisStreamMessageHandler streamMessageHandler = new ReactiveRedisStreamMessageHandler(streamKey, redisConnectionFactory);
+		streamMessageHandler.setSerializationContext(jsonSerializationContext);
+		//initializes reactiveRedisStreamOperations
+		invokeOnInitMethod(streamMessageHandler);
+
+		ReactiveMessageHandlerAdapter handlerAdapter = new ReactiveMessageHandlerAdapter(streamMessageHandler);
+		handlerAdapter.handleMessage(message);
+
+		ReactiveRedisTemplate<String, String> template = new ReactiveRedisTemplate(redisConnectionFactory, jsonSerializationContext);
+		ObjectRecord<String, RedisMessageStoreTests.Person> record = template.opsForStream().read(RedisMessageStoreTests.Person.class, StreamOffset.fromStart(streamKey)).blockFirst();
```

Comment:
That's not good. 
If you think that you rally need that `Person` for your goal, then consider to move it to the top-level package, e.g. `org.springframework.integration.redis.util`.